### ISSUE TITLE
Windows move offscreen slightly when tiled to certain corners

### ISSUE THREAD

**Issue by pwfrank** [linuxmint/muffin#396]
```
 * muffin version 4.0.5
 * Distribution - Arch
 * Cinnamon version 4.0.6
 * Intel HD Graphics 4000, proprietary driver
 * 64 bit
 ```

**Issue**
Windows are pushed offscreen slightly when tiled from the E edge to the NE corner of the screen using keyboard shortcuts. Also occurs when a window on the S edge is tiled to the SW corner. 

**Steps to reproduce**
Tile or snap a window to cover the right half of the screen, then use a "tiling and snapping" keyboard shortcut to tile the window upwards. The window will be pushed offscreen to the right by about 10-15 pixels.

**Expected behaviour**
Windows snapped into a corner from any position should remain entirely within the display area, flush against any neighboring windows.

**Other information**
- Windows must have a Cinnamon window title bar to display this behavior. E.g. Chromium windows tile improperly only when using the "use system title bar and borders" option

- Behavior occurs with both the "Push snap" and "Push tile" shortcuts

- Windows still tile properly when dragged to the corner

- Occurs on both the laptop internal display and an external monitor

Two tiled windows, each covering half the screen
![improper_tiling1](https://user-images.githubusercontent.com/33354358/49970611-e78eb000-fef9-11e8-9640-b670c3e0b7a4.png)

Right window pushed slightly offscreen to the right after tiling upwards
![improper_tiling2](https://user-images.githubusercontent.com/33354358/49970628-f70df900-fef9-11e8-8811-05f4ddf1e02b.png)

**Comment by pwfrank** [comment]
This issue seems related to a previous fix, (https://github.com/linuxmint/muffin/issues/154). But since this issue only occurs with windows that have a title bar, I think the bug is most likely in the way the `new_x` and `new_y` variables in the `handle_move_to_corner_backend()` function are calculated for these windows

**Comment by Axelderan** [comment]
I have a video of the same thing: https://drive.google.com/open?id=1P2Jhivx-rK8xxEY5q3J4xrcltd6uzf0W

```
System:
  Host: absurd-reckonings Kernel: 4.20.0-1-MANJARO x86_64 bits: 64 
  compiler: gcc v: 8.2.1 Desktop: Cinnamon 4.0.8 dm: LightDM 1.28.0 
  Distro: Manjaro Linux 
Graphics:
  Device-1: Intel UHD Graphics 620 vendor: Hewlett-Packard driver: i915 
  v: kernel bus ID: 00:02.0 chip ID: 8086:5917 
  Display: x11 server: X.Org 1.20.3 driver: intel unloaded: modesetting 
  alternate: fbdev,vesa tty: N/A 
  OpenGL: renderer: Mesa DRI Intel UHD Graphics 620 (Kabylake GT2) 
  v: 4.5 Mesa 18.3.1 compat-v: 3.0 direct render: Yes 
```

**Comment by jaszhix** [comment]
This is fixed in #411.
